client/src/main/java/org/asynchttpclient/netty/channel/ChannelManager.java

```diff
@@ -488,4 +493,28 @@ public ChannelPool getChannelPool() {
     public EventLoopGroup getEventLoopGroup() {
         return eventLoopGroup;
     }
+
+    public ClientStats getClientStats() {
+        final Map<String, Long> totalConnectionsPerHost = openChannels
+                    .stream()
+                    .map(Channel::remoteAddress)
+                    .filter(a -> a.getClass() == InetSocketAddress.class)
+                    .map(a -> (InetSocketAddress) a)
```

Comment:
So this sucks, but I didn't really see another way out. Are there any other types of socket that are likely to live here or in `openChannels`, or is this a mostly "safe" thing to do?